sys/netlink/netlink_message_parser.h

```diff
@@ -47,11 +50,21 @@ static inline void *
 lb_alloc(struct linear_buffer *lb, int len)
 {
 	len = roundup2(len, _Alignof(__max_align_t));
+#ifdef __CHERI_PURE_CAPABILITY__
+	len = CHERI_REPRESENTABLE_LENGTH(len);
+	char *data = __align_up(lb->base + lb->offset,
+	    CHERI_REPRESENTABLE_ALIGNMENT(len));
+	if (data + len > lb->base + lb->size)
+		return (NULL);
+	lb->offset = (data + len) - lb->base;
+	return (cheri_setboundsexact(data, len));
```

Comment:
Can't we use the same logic for both and let the compiler elide the unncessary logic?